File: backend/src/main/java/com/yourorg/appname/service/MatchService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.CreateMatchRequest;
import com.yourorg.appname.dto.request.RecordMatchResultRequest;
import com.yourorg.appname.dto.response.MatchResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public interface MatchService {

    MatchResponse createMatch(CreateMatchRequest request);

    MatchResponse recordMatchResult(Long matchId, RecordMatchResultRequest request);

    MatchResponse recordQuickResult(RecordMatchResultRequest request);

    List<MatchResponse> getRecentMatches(int limit);

    Page<MatchResponse> getUserMatchHistory(int page, int size);

    MatchResponse getLastEncounter();
}

File: frontend/src/services/matchService.js
import apiClient from './apiClient';

export const matchService = {
  async createMatch(matchData) {
    const response = await apiClient.post('/matches', matchData);
    return response.data?.data;
  },

  async recordResult(matchId, resultData) {
    const response = await apiClient.post(`/matches/${matchId}/result`, resultData);
    return response.data?.data;
  },

  async recordQuickResult(resultData) {
    const response = await apiClient.post('/matches/quick-result', resultData);
    return response.data?.data;
  },

  async getRecentMatches(limit = 5) {
    const response = await apiClient.get(`/matches/recent?limit=${limit}`);
    return response.data?.data;
  },

  async getLastEncounter() {
    const response = await apiClient.get('/matches/last-encounter');
    return response.data?.data;
  },

  async getMatchHistory(page = 0, size = 10) {
    const response = await apiClient.get(`/matches/history?page=${page}&size=${size}`);
    return response.data?.data;
  },
};
